apps/openssl.c

```diff
@@ -787,18 +793,23 @@ static void list_disabled(void)
 
 static LHASH_OF(FUNCTION) *prog_init(void)
 {
-    LHASH_OF(FUNCTION) *ret;
+    static LHASH_OF(FUNCTION) *ret = NULL;
+    static int prog_inited = 0;
     FUNCTION *f;
     size_t i;
 
-    /* Sort alphabetically within category. For nicer help displays. */
-    for (i = 0, f = functions; f->name != NULL; ++f, ++i) ;
-    qsort(functions, i, sizeof(*functions), SortFnByName);
+    if (!prog_inited) {
```

Comment:
you don't need prog_inited.  just do
``
if (ret != NULL)
    return ret;
```
And then don't indent the rest of the function body.

Reply:
So if it fails creating the table, it should retry the next time this is called?  That flag is an indicator that it was tried once, don't even bother trying again

Reply:
... however, I can still change the code around to avoid all that extra indentation ;-)